scripts/brands-page/brands.js

```diff
@@ -0,0 +1,291 @@
+/**
+ * Fetches a list of brands from the API and renders the result into the 'brandsList' element.
+ *
+ * On success, the fetched brand data is passed to the `renderBrands` function to display the list.
+ * On failure, an error is logged to the console and the `renderError` function is called to display an error message.
+ *
+ * @async
+ * @throws {Error} Throws an error if the API response is not successful.
+ */
+
+let currentPage = 1
+let currentSortColumn = 'id'
+let currentSortOrder = 'ASCENDING'
+let filters = [] // Array to hold filter objects
+const renderTable = (brands) => {
+    const tableBody = document.getElementById('brandsTableBody')
+    if (!tableBody) {
+        console.error('Table body element not found')
+        return
+    }
+
+    // Clear existing rows
+    tableBody.innerHTML = ''
+
+    // Add new rows
+    tableBody.innerHTML = brands
+        .map(
+            (brand) => `
+            <tr>
+                <td><input type="checkbox" class="selectCheckbox" data-id="${brand.id}"></td>
+                <td>${brand.id}</td>
+                <td>${brand.name}</td>
+                <td><button class="btn btn-sm btn-primary">Edit</button></td>
+            </tr>
+        `
+        )
+        .join('')
+}
+
+let isFetching = false
+
+const fetchAndRenderBrands = async () => {
+    if (isFetching) return // Prevent new requests while a fetch is ongoing
+    isFetching = true
+
+    const API_URL = 'https://core-995b.onrender.com/admin/brands/search'
+    const requestBody = {
+        page: currentPage,
+        size: 10,
+        sorts: [
+            {
+                field: currentSortColumn,
+                direction: currentSortOrder
+            }
+        ],
+        filters
+    }
+
+    console.log('Requesting with payload:', JSON.stringify(requestBody, null, 2))
+
+    try {
+        const response = await fetch(API_URL, {
+            method: 'POST',
+            headers: { 'Content-Type': 'application/json' },
+            body: JSON.stringify(requestBody)
+        })
+
+        if (!response.ok) {
+            throw new Error(`Error: ${response.status} ${response.statusText}`)
+        }
+
+        const data = await response.json()
```

Comment:
What do you think about adding type definition? Is it a good practice or useless? Yeah, I know TypeScript exists))

/**
* @typedef {Object} SearchResponse
* @property {Array} data
* @property {number} page
* @property {number} size
* @property {number} totalElements
*/

/** @type {SearchResponse} */